adafruit_circuitplayground/express.py

```diff
@@ -569,9 +562,10 @@ def _sine_sample(length):
         for i in range(length):
             yield int(tone_volume * math.sin(2*math.pi*(i / length)) + shift)
 
-    def _generate_sample(self, length=100):
+    def _generate_sample(self):
```

Comment:
`length` should remain a kwarg, and not be hard-coded in `_generate_sample()`. In `start_tone()`, length could be a number other than 100. Starting at line # 625:
```python
self._speaker_enable.value = True
length = 100
if length * frequency > 350000:
    length = 350000 // frequency
self._generate_sample(length)
```

Reply:
I may need more coffee today. I see that you removed that section in `start_tone()`, but I still don't understand the why behind it.

Reply:
to remove length param from _generateh_sample() when need to remove it from _sine_sample() too, then add the math for it to _sine_sample() super easy just '100'...

Reply:
dargh nvm i see what your saying i mistakenly removed it for no reason, even though its not used ever